src/Cryptol/TypeCheck/Solve.hs

```diff
@@ -290,10 +290,10 @@ proveImplication dedupErrs lnam as ps gs =
 -- | Tries to prove an implication. If proved, then returns `Right (m_su ::
 -- InferM Subst)` where `m_su` is an `InferM` computation that results in the
```

Comment:
This claims that it returns `Right (m_su :: InferM Subst)`, which doesn't seem right. I wonder if this is a leftover from a previous implementation. I'd favor removing the mention of the substitution from the Haddocks and instead changing it to something like "return `Right warnings`, where `warnings` records all the warnings invoked during proving".